src/Lumi/Components2/Banner.purs

```diff
@@ -0,0 +1,155 @@
+  ( Banner
+  , BannerProps
+  , banner
+  , actionBanner
+  , primary
+  , active
+  , warning
+  , error
+  , module Styles.Banner
+  ) where
+
+import Prelude
+
+import Color (lighten)
+import Data.Foldable (fold)
+import Data.Monoid as Monoid
+import Data.Tuple.Nested ((/\))
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, PropsModifier, lumiComponent, lumiElement)
+import Lumi.Components.Icon as Icon
+import Lumi.Components.Spacing (Space(..))
+import Lumi.Components2.Box (box)
+import Lumi.Styles (color, css, nested, prop, str, styleModifier, styleModifier_)
+import Lumi.Styles.Banner (banner) as S
+import Lumi.Styles.Border (_listSpaced, _listCompact) as Styles.Banner
+import Lumi.Styles.Box (FlexAlign(..), _align, _column, _fill, _interactive, _justify, _row) as S
+import Lumi.Styles.Responsive (desktopQuery)
+import Lumi.Styles.Responsive (onDesktop) as S
+import Lumi.Styles.Theme (LumiTheme(..), useTheme)
+import React.Basic (JSX)
+import React.Basic.DOM as R
+import React.Basic.DOM.Events (capture_)
+import React.Basic.Hooks (useState)
+import React.Basic.Hooks as React
+
+data Banner = Banner
+
+type BannerProps =
+  ( component :: Banner
+  , dismissable :: Boolean
+  , content :: Array JSX
+  )
+
+banner :: LumiComponent BannerProps
+banner =
+  unsafePerformEffect do
+    lumiComponent "Banner" defaults \props -> React.do
+      theme <- useTheme
+      visible /\ setVisible <- useState true
+      pure
+        $ Monoid.guard visible
+        $ lumiElement box
+        $ styleModifier props.css
+        $ S.banner
+        $ _ { className = props.className
+            , content =
+                [ lumiElement box
+                  $ S.onDesktop S._row
+                  $ S._align S.Center
+                  $ S._fill
+                  $ _ { content = props.content }
+                , Monoid.guard props.dismissable
+                    $ lumiElement box
+                    $ dismissButtonStyle
+                    $ _ { content =
+                            [ Icon.icon
+                                { style: R.css { fontSize: "12px" }
+                                , type_: Icon.Remove
+                                }
+                            ]
+                        , onClick = capture_ $ setVisible \_ -> false
+                        }
+                ]
+            }
+  where
+    defaults :: Record BannerProps
+    defaults =
+      { component: Banner
+      , content: []
+      , dismissable: false
+      }
+
+    dismissButtonStyle :: forall props. PropsModifier props
+    dismissButtonStyle =
+      S._interactive
+      >>> styleModifier \(LumiTheme { colors }) ->
+            css
+              { padding: str "8px"
+              , margin: str "0 -8px 0 8px"
+              , "&:hover": nested $ css
+                  { color: color colors.black1
+                  }
+              }
+
+actionBanner :: Array JSX -> PropsModifier BannerProps
+actionBanner actions f =
```

Comment:
I wonder if this would be better as a `LumiComponent` instead of a `PropsModifier`.

Reply:
This does seem like a strange way to use it.. hmm

Reply:
Yeah, I think so too. Would you prefer having it as a `LumiComponent`?